File: src/test/java/com/baulsupp/oksocial/output/ConsoleHandlerTest.java
package com.baulsupp.oksocial.output;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.dataformat.cbor.CBORFactory;
import java.io.ByteArrayInputStream;
import java.io.File;
import java.io.IOException;
import java.util.LinkedHashMap;
import java.util.Map;
import java.util.Optional;
import okio.BufferedSource;
import okio.Okio;
import org.junit.Test;

import static java.util.Optional.of;

public class ConsoleHandlerTest {
  @Test
  public void testCborSupport() throws IOException {
    ObjectMapper cborMapper = new ObjectMapper(new CBORFactory());

    Map<String, String> s = new LinkedHashMap<String, String>();
    s.put("a", "AAA");
    s.put("b", "BBB");
    byte[] bytes = cborMapper.writeValueAsBytes(s);

    ResponseExtractor<byte[]> extractor = new ResponseExtractor<byte[]>() {
      @Override public Optional<String> mimeType(byte[] response) {
        return of("application/cbor");
      }

      @Override public BufferedSource source(byte[] response) throws IOException {
        return Okio.buffer(Okio.source(new ByteArrayInputStream(response)));
      }

      @Override public String filename(byte[] response) {
        return "filename";
      }
    };
    ConsoleHandler<byte[]> c = new ConsoleHandler<>(extractor);
    c.showOutput(bytes);
  }
}

File: src/test/kotlin/com/baulsupp/oksocial/output/ConsoleHandlerTest.kt
package com.baulsupp.oksocial.output

import com.fasterxml.jackson.databind.ObjectMapper
import com.fasterxml.jackson.dataformat.cbor.CBORFactory
import kotlinx.coroutines.runBlocking
import okio.BufferedSource
import okio.buffer
import okio.source
import org.junit.Test
import java.io.ByteArrayInputStream
import java.util.LinkedHashMap

class ConsoleHandlerTest {
  @Test
  fun testCborSupport() {
    val cborMapper = ObjectMapper(CBORFactory())

    val s = LinkedHashMap<String, String>()
    s["a"] = "AAA"
    s["b"] = "BBB"
    val bytes = cborMapper.writeValueAsBytes(s)

    val extractor = object : ResponseExtractor<ByteArray> {
      override fun mimeType(response: ByteArray): String? {
        return "application/cbor"
      }

      override fun source(response: ByteArray): BufferedSource {
        return ByteArrayInputStream(response).source().buffer()
      }

      override fun filename(response: ByteArray): String {
        return "filename"
      }
    }
    val c = ConsoleHandler(extractor)
    runBlocking {
      c.showOutput(bytes)
    }
  }
}
